---
title: "Lab 08"
author: "Jack Reddan"
date: "10/22/2021"
#output: github_document
output: pdf_document
---

## Import the Data

```{r}
url <- "https://tinyurl.com/UK-foods"
uk_foods <- read.csv(url, row.names = 1)
```

## Explore Imported Data
>Question 1: How many rows and columns are in your new data frame named x? What R functions could you use to answer this questions?  

```{r}
dim(uk_foods)
```
  
17 rows and 4 columns (*17 food categories and 4 countries*)  

## Checking the Data

```{r}
head(uk_foods)
```

>Question 2: Which approach to solving the ‘row-names problem’ mentioned above do you prefer and why? Is one approach more robust than another under certain circumstances?  

I prefer to read in the row-names when I read in the csv data (in *read.csv*).
This is less dangerous since it does not mutate our data in the body of the 
code, and instead restricts it to when the data is read in.

# Looking for differences and trends

```{r}
barplot(as.matrix(uk_foods), beside = T, col = rainbow(nrow(uk_foods)))
```
  
>Question 3: Changing what optional argument in the above barplot() function results in the following plot?  

```{r, echo=FALSE}
barplot(as.matrix(uk_foods), beside = F, col = rainbow(nrow(uk_foods)))
```  
  
Changing the `beside` argument from `TRUE` to `FALSE`  

>Question 5: Generating all pairwise plots may help somewhat. Can you make sense of the following code and resulting figure? What does it mean if a given point lies on the diagonal for a given plot?  

```{r}
pairs(uk_foods, col=rainbow(10), pch=16)
```  

This code plots all pairwise comparisons of row values across the column groups.
If a given point lies on the diagonal, this means that the value of the
corresponding row is the same in both columns which are plotted. A point above
the diagonal would be higher in the column plotted on the y-axis than in the column plotted on the x-axis. A point below the diagonal would be higher in the 
column plotted on the x-axis than in the column plotted on the y-axis.  

>Question 6: What is the main differences between N. Ireland and the other countries of the UK in terms of this data-set?  

>>Without PCA 

This is difficult to do without using a PCA, but a way could be to look at the maximum differences, pairwise, in log2 fold-changes between Northern Ireland and the other UK countries.

Identify the log2 fold change and identify the component with the maximum 
absolute change
```{r}
n_ireland_v_england <- log(uk_foods$N.Ireland, base = 2)/log(uk_foods$England, base = 2)
n_ireland_v_wales <- log(uk_foods$N.Ireland, base = 2)/log(uk_foods$Wales, base = 2)
n_ireland_v_scotland <- log(uk_foods$N.Ireland, base = 2)/log(uk_foods$Scotland, base = 2)

names(n_ireland_v_england) <- row.names(uk_foods)
names(n_ireland_v_wales) <- row.names(uk_foods)
names(n_ireland_v_scotland) <- row.names(uk_foods)

which.max(abs(n_ireland_v_england))
which.max(abs(n_ireland_v_wales))
which.max(abs(n_ireland_v_scotland))
```


## PCA

### Conduct PCA with the UKFoods data set
```{r}
uk_foods_pca <- prcomp(t(uk_foods))
summary(uk_foods_pca)
```

### Inspect PCAs 
```{r}
plot(uk_foods_pca)
```

### Plot the data using PCs  
  
Use the two PCs which explain the most variance to plot the UKFoods data set (PC1
and PC2)  

>Question 7: 
  
```{r}
plot(uk_foods_pca$x[,1:2])
text(uk_foods_pca$x[,1:2], colnames(uk_foods))
```
  
>Question 8: Customize your plot so that the colors of the country names match 
the colors in our UK and Ireland map and table at start of this document.

```{r}
color_vector <- c("orange", "red", "blue", "green")
plot(uk_foods_pca$x[,1:2], col = "#DDDDDD", xlim = c(-300, 600))
text(uk_foods_pca$x[,1:2], colnames(uk_foods), col = color_vector)
```
  
### Plot the proportions of variences explained by each PC
```{r}
variance_per <- round(((uk_foods_pca$sdev^2)/sum(uk_foods_pca$sdev^2)) * 100)

barplot(variance_per, xlab="Principal Component", ylab="Percent Variation")
```
### Digging Deeper: Variable Loadings

```{r}
uk_foods_pca$rotation
```
  
Since PC1 accounts for the most variance, the contribution of each component
on the distribution of countries on this PC will be the most helpful.  
```{r}
op <- par(mar=c(10,3,0.35,0))
barplot(uk_foods_pca$rotation[,1], las = 2)
par(op)
```
  
>Question 9: Generate a similar ‘loadings plot’ for PC2. What two food groups feature prominantely and what does PC2 maninly tell us about?
  
```{r}
op <- par(mar=c(10,3,0.35,0))
barplot(uk_foods_pca$rotation[,2], las = 2)
par(op)
```
  
The two most prominent features are Fresh potatoes and soft drinks. This PC is mainly telling us which features most contribute to the differences between the 
other three UK countries (England to Scotland to Wales) since it is on this PC
that these countries are differentiated upon. Therefore, these features (fresh potatoes and soft drinks) are the components which contribute to the differences
among these three countries.  
  
#PCA of RNA-Seq Data

## Import the Data
```{r}
url2 <- "https://tinyurl.com/expression-CSV"
rna_data <- read.csv(url2, row.names=1)
head(rna_data)
```

>Question 10: How many genes and samples are in this data set?  

```{r}
dim(rna_data)
```
  
100 genes and 10 samples in the data set.
  
## PCA for RNA-Seq Data
```{r}
rna_pca <- prcomp(t(rna_data), scale = TRUE)
summary(rna_pca)
```

Scree plot of the PCs helps to show how much variance is explained by each PC.
Here it is clear that PC1 dominates compared to the others,
explaining >92% of the total variance.
  
```{r}
plot(rna_pca, main = "Scree plot")
```

Replicate the Scree plot method for the pca object with base R plotting.
  
```{r}
rna_pca_variance <- rna_pca$sdev^2
rna_pca_varience_percent <- round(rna_pca_variance/sum(rna_pca_variance)*100, 1)

barplot(rna_pca_varience_percent, main="Mimicked Scree Plot", 
        names.arg = paste0("PC", 1:10),
        ylim = c(0, 100),
        xlab="Principal Component", 
        ylab="Percent Variation")
```
  
Plot the data using PC1 and PC2
```{r}
plot(rna_pca$x[,1], rna_pca$x[,2], xlab = "PC1", ylab = "PC2")
```

Check if the groupings make sense
```{r}
color_vector <- colnames(rna_data)
color_vector[grep("wt", color_vector)] <- "red"
color_vector[grep("ko", color_vector)] <- "blue"

plot(rna_pca$x[,1], rna_pca$x[,2], col=color_vector, pch=16,
     xlab=paste0("PC1 (", rna_pca_varience_percent[1], "%)"),
     ylab=paste0("PC2 (", rna_pca_varience_percent[2], "%)"))

text(rna_pca$x[,1], rna_pca$x[,2], labels = colnames(rna_data), pos=c(rep(4,5), rep(2,5)))
```
  
### Plotting PCA using ggplot

```{r}
# Load the ggplot2 library
library(ggplot2)

# Convert PCA results to a data frame for ggplot
rna_pca_df <- as.data.frame(rna_pca$x)

# Plot samples on PC1 and PC2
ggplot(rna_pca_df) +
  aes(x = PC1, y = PC2) +
  geom_point()
```
  
To make the figure clearer, plot the labels instead of just a point, and map
color to sample condition.

```{r}
# Create columns in the data frame for labeling samples and conditions
rna_pca_df$samples <- colnames(rna_data)
rna_pca_df$conditions <- substr(colnames(rna_data),1,2)

ggplot(data = rna_pca_df) +
  aes(x = PC1, y = PC2, label = samples, color = conditions) +
  geom_label(show.legend = FALSE)
```
  
>Refine the ggplot figure with themes and labels
  
```{r}
ggplot(data = rna_pca_df) +
  aes(x = PC1, y = PC2, label = samples, color = conditions) +
  geom_label(show.legend = FALSE) +
  labs(
    title = "PCA of RNA-sequencing data",
    x = paste0("PC1 (", rna_pca_varience_percent[1], "%)"),
    y = paste0("PC2 (", rna_pca_varience_percent[2], "%)"),
    caption = "BGGN 213 example data (AY21 F)") +
  theme_minimal()
```
  
### Gene loadings

Explore which genes contribute most to PC1 to determine what the key, differentiating genes are between the two condition groups.
  
```{r}
# Save gene loading scores for PC1
gene_loading_scores <- rna_pca$rotation[,1]

# Rank by absolute value from high to low
ranked_absolute_loading_scores <- sort(abs(gene_loading_scores), decreasing=TRUE)

# Print the gene names with the first 10 (top 10) scores
names(ranked_absolute_loading_scores[1:10])
```

## Session Info  
```{r}
sessionInfo()
```